examples/embedding/main_test.go

```diff
@@ -5,58 +5,12 @@
 package main
 
 import (
-	"bytes"
-	"flag"
-	"os"
-	"os/exec"
-	"path/filepath"
 	"testing"
-)
 
-var regen = flag.Bool("regen", false, "regenerate golden files")
+	"github.com/go-python/gpython/pytest"
+)
 
 func TestEmbeddedExample(t *testing.T) {
-
-	tmp, err := os.MkdirTemp("", "go-python-embedding-")
-	if err != nil {
-		t.Fatal(err)
-	}
-	defer os.RemoveAll(tmp)
-
-	exe := filepath.Join(tmp, "out.exe")
-	cmd := exec.Command("go", "build", "-o", exe, ".")
-	err = cmd.Run()
-	if err != nil {
-		t.Fatalf("failed to compile embedding example: %+v", err)
-	}
-
-	out := new(bytes.Buffer)
-	cmd = exec.Command(exe, "mylib-demo.py")
-	cmd.Stdout = out
-	cmd.Stderr = out
-
-	err = cmd.Run()
-	if err != nil {
-		t.Fatalf("failed to run embedding binary: %+v", err)
-	}
-
-	const fname = "testdata/embedding_out_golden.txt"
-
-	got := out.Bytes()
-
-	flag.Parse()
-	if *regen {
-		err = os.WriteFile(fname, got, 0644)
-		if err != nil {
-			t.Fatalf("could not write golden file: %+v", err)
-		}
-	}
-
-	want, err := os.ReadFile(fname)
-	if err != nil {
-		t.Fatalf("could not read golden file: %+v", err)
-	}
-	if !bytes.Equal(got, want) {
-		t.Fatalf("stdout differ:\ngot:\n%s\nwant:\n%s\n", got, want)
-	}
+	//pytest.RegenGoldFiles = true
+	pytest.RunScript(t, "./mylib-demo.py")
```

Comment:
replacing `TestEmbeddedExample()` with, effectively, `pytest.RunScript(...)` seems to completely discard compiling a customized `gpython` interpreter with the `mylib` module, doesn't it?

if my understanding is correct, then we probably have a problem with that test succeeding :}

Reply:
Hi friend!  Hope you are well in this crazy world.  :\

Hmm, trying to understand you question here.

My intent with this PR was to more expose and use the "_golden.txt" code you added a couple months ago in order to:
   - drop the (now redundant) golden code in the mylib example
   - lay the groundwork (for a PR to follow) that runs the multi-context example also in the form of a test (as we had previously discussed)

Perhaps to your point, indeed, running `mylib` in a test (vs its `main.go`) is essentially the same.  So if you are saying what is the point of having `main.go`, I would say it's to demonstrate and provide an example that newcomers can relate to, drill into, and not get intimidated.  This is based on my assumption that others are like me and tend to look for things like `main.go` as a starting point (and not tests).   My logic centers around making it as easy and clear as possible for newcomers to explore a completely stand-alone embedded example.

Or perhaps I am misunderstanding your question?

Separately...

Yes, stuff like `pytest.RegenGoldFiles = true` is annoying to have around.  It sounds like the ` flag.Bool(..., false, ...)` approach you described is best.

Reply:
let me restate my question then.

in the new version of `TestEmbeddedExample`, where is the code of `mylib` coming from?
`mylib-demo.py` imports `mylib`.
and `mylib` imports, in turn, `mylib_go`.

in the old version, `mylib_go` was coming from the compilation of `mylib.module.go` together with `gpython`'s `main`.
now, where is it coming from?